src/main/java/com/zerobase/cafekiosk/cart/service/CartServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+package com.zerobase.cafekiosk.cart.service;
+
+import com.zerobase.cafekiosk.beverage.entity.Beverage;
+import com.zerobase.cafekiosk.beverage.repository.BeverageRepository;
+import com.zerobase.cafekiosk.cart.constant.CartStatus;
+import com.zerobase.cafekiosk.cart.dto.CartDto;
+import com.zerobase.cafekiosk.cart.entity.Cart;
+import com.zerobase.cafekiosk.cart.model.CartInput;
+import com.zerobase.cafekiosk.cart.repository.CartRepository;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CartServiceImpl implements CartService {
+
+  private final CartRepository cartRepository;
+  private final BeverageRepository beverageRepository;
+
+  public static int calculatePrice(CartDto cartDto) {
+    return cartDto.getQuantity() * cartDto.getPrice();
+  }
+
+  @Override
+  public List<CartDto> list() {
+
+    List<Cart> carts = cartRepository.findAll();
+
+    return CartDto.of(carts);
+  }
+
+  @Override
+  public CartDto add(CartInput request) {
+
+    Optional<Cart> optionalCart = cartRepository.findByBeverageIdAndCartStatus
+        (request.getBeverageId(), CartStatus.ORDERED);
+
+    if (optionalCart.isPresent()) {
+      Cart cart = optionalCart.get();
+      cart.setQuantity(cart.getQuantity() + request.getQuantity());
+      cartRepository.save(cart);
+      return CartDto.of(cart);
+    } else {
+      Beverage beverage = beverageRepository.findById(request.getBeverageId())
+          .orElseThrow(() -> new RuntimeException("존재하지 않는 음료입니다."));
+      request.setBeverageName(beverage.getBeverageName());
+      request.setPrice(beverage.getPrice());
+      Cart cart = new Cart().buildCart(request);
```

Comment:
Cart를 만들때 request를 이용해서 만드는데요, 위에서 사용자가 입력하지 않은 데이터인 price, beverageName을 같이 넣어주는 것 같아요. 이 부분은 request에 넣어주기보단 따로 넣어주시는 것이 코드 구조상 더 맞을 것 같네요
```suggestion
      Cart cart = new Cart().buildCart(request, beverageName, price);
```